runtime-watcher/tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,237 @@
+package e2e_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/kyma-project/runtime-watcher/skr/tests/e2e/utils"
+	apicorev1 "k8s.io/api/core/v1"
+
+	apiappsv1 "k8s.io/api/apps/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	controlPlaneNamespace = "kcp-system"
+	watcherCrName         = "klm-kyma-watcher"
+	remoteNamespace       = "kyma-system"
+	localHostname         = "0.0.0.0"
+	k3dHostname           = "host.k3d.internal"
+	kymaName              = "kyma-sample"
+	kymaChannel           = "regular"
+	watcherSecretName     = "skr-webhook-tls" //nolint:gosec
+	watcherName           = "skr-webhook"
+)
+
+type ResourceName = types.NamespacedName
+
+var errWatcherDeploymentNotReady = errors.New("watcher Deployment is not ready")
+
+var _ = Describe("Enqueue Event from Watcher", Ordered, func() {
+	kyma := utils.NewKyma(kymaName, controlPlaneNamespace, kymaChannel,
+		v1beta2.SyncStrategyLocalSecret)
+	GinkgoWriter.Printf("kyma before create %v\n", kyma)
+	incomingRequestMsg := fmt.Sprintf("event received from SKR, adding %s/%s to queue",
+		kyma.GetNamespace(), kyma.GetName())
+
+	initEmptyKymaBeforeAll(kyma)
+	cleanupKymaAfterAll(kyma)
+
+	skrSecret := ResourceName{
+		Name:      watcherSecretName,
+		Namespace: remoteNamespace,
+	}
+	watcher := ResourceName{
+		Namespace: remoteNamespace,
+		Name:      watcherName,
+	}
+
+	Context("Given SKR Cluster with TLS Secret", func() {
```

Comment:
@ameteiko Do you have any improvements for the main test steps arrangement of this Context body?

Reply:
@lindnerby It looks consistent, readable and straightforward as it is already, great job! 

To verify it, you can use `make e2e-coverage` from the `/test/e2e` directory to see the resulting report.

Reply:
Alright, thanks!